components/MintControls/index.tsx

```diff
@@ -49,10 +50,12 @@ export default function MintControls({ pageState, setPageState }: MintControlsPr
     return cost.toString()
   }, [mintState, isEarlyMinter])
 
-  const contractAddress = useMemo(
-    () => VALENFTINES_ADDRESS[(network.chain?.id as SupportedChainId) || SupportedChainId.MAINNET],
-    [network]
-  )
+  const contractAddress = useMemo(() => {
+    if (!onSupportedNetwork) {
+      return VALENFTINES_ADDRESS[SupportedChainId.MAINNET]
+    }
+    return VALENFTINES_ADDRESS[network.chain?.id as SupportedChainId]
+  }, [network.chain?.id, onSupportedNetwork])
```

Comment:
low priority, but I want to get a graceful fallback going on unsupported networks